src/nemos/utils.py

```diff
@@ -63,7 +63,7 @@ def check_dimensionality(
 def convolve_1d_trials(
     basis_matrix: ArrayLike,
     time_series: Union[Iterable[NDArray], NDArray, Iterable[jnp.ndarray], jnp.ndarray],
-) -> List[jnp.ndarray]:
+) -> Union[Iterable[NDArray], NDArray, Iterable[jnp.ndarray], jnp.ndarray]:
```

Comment:
I'm confused by the `Iterable` in the type hinting here (both for return and `time_series`). Why do that instead of List?

Reply:
Is this the current best practice for pytree annotation? That feels ... not great. Maybe we can be more specific about this? As in, we look for how we use it in our docs and library, and set it to that. Or make it clear somewhere else?

Reply:
I guess the docstring can be more specific